src/modules/0.0.22/aws_s3/rpcs/s3_clean_bucket.ts

```diff
@@ -38,29 +45,58 @@ export class S3CleanBucketRpc extends RpcBase {
         },
       ];
     }
-    const client = (await ctx.getAwsClient()) as AWS;
-    const objects = await this.getBucketObjects(client.s3Client, bucketName);
-    for (const object of objects) {
-      // delete the object
-      await this.deleteBucketObject(client.s3Client, bucketName, object.Key);
-    }
+    const client = (await ctx.getAwsClient(await ctx.getDefaultRegion())) as AWS;
 
-    // query again to see if all objects have been deleted
-    const remainingObjects = await this.getBucketObjects(client.s3Client, bucketName);
-    if (!remainingObjects.length) {
-      return [
-        {
-          bucket: bucketName,
-          status: 'OK',
-          response_message: 'All bucket objects have been deleted',
-        },
-      ];
+    // first determine bucket region
+    let region = await this.getBucketLocation(client.s3Client, bucketName);
+    if (!region) region = 'us-east-1';
+    const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
+
+    if (region) {
+      // check if it is on enabled regions
+      if (enabledRegions.includes(region)) {
+        const client1 = (await ctx.getAwsClient(region)) as AWS;
```

Comment:
Why is this called `client1`? I don't see a `client2`?

Reply:
There was a client, using the default zone, and then i create another client, specific for the bucket region, to clean the bucket